docs/source/considerations.rst

```diff
@@ -143,6 +143,10 @@ however you may notice some differences:
 #. When setting a handler that is a method already defined in the agent we
    simply pass a string with the method name.
 
+.. note:: The ``configure()`` method is preferred over ``on_init()``, since
+   both will be executed before the main loop, but ``on_init()`` will be
+   called before the agent is running, which can lead to some unexpected errors.
+
```

Comment:
Rather than preferred... I would simply state in the note that `on_init()` exists and why is it different.

In example:

"While `configure()` is executed right before the main loop (i.e.: when the agent is already running), we might want to use the lower-level `on_init()` method, which is executed right after `Agent`'s initialization."

You could also remove the paragraph all together and let the user know only about the `.configure()` method... We might even deprecate/remove `.on_init()` in the future if we find out it is not necessary.

Reply:
I'd rather keep the note for now. As you said, we can remove `on_init` in the future if turns out not to be necessary after all.

Reply:
I would remove this paragraph now. Nothing changed and `on_init` should still be used for initial attributes setting and socket binding. We added a new method, but that is explained in another section (in the timers section).